beacon_node/http_api/src/lib.rs

```diff
@@ -1933,6 +1933,77 @@ pub fn serve<T: BeaconChainTypes>(
             })
         });
 
+    /*
+     * builder/states
+     */
+
+    let builder_states_path = eth_v1
+        .and(warp::path("builder"))
+        .and(warp::path("states"))
+        .and(chain_filter.clone());
+
+    // GET builder/states/{state_id}/expected_withdrawals
+    let get_expected_withdrawals = builder_states_path
+        .clone()
+        .and(warp::path::param::<StateId>())
+        .and(warp::query::<api_types::ExpectedWithdrawalsQuery>())
```

Comment:
I think `query` generally gets added after `path` params - I'm not 100% sure whether the order matters with `warp`, this might still work but I think would be a bit more consistent and readable to have it at the end.